### PR TITLE
Only accept return_url when do_update_woocommerce is present

### LINKED ISSUE #48154: [9.0.0 RC1]: All links containing "return_url" query var are broken
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

As soon as you have the `return_url` query var in an admin URL, you are automatically redirected to this "return URL".

This bug has been caused by this commit => https://github.com/woocommerce/woocommerce/commit/da57287c560ab8b4d9fc3f0f80b3fc54f2944e21

The scope of the logic in `WC_Install::install_actions` should be reduced.

### Expected behavior

WooCommerce should not redirect all requests based on the `return_url` query var:

Example:
https://wpml.loc/wp-admin/post.php?post=1&action=edit&return_url=https://wpml.loc/ should just open the post editor.

### Actual behavior

Example:
https://wpml.loc/wp-admin/post.php?post=1&action=edit&return_url=https://wpml.loc/ is redirecting us to https://wpml.loc/ (the home page)

### Steps to reproduce

1. On a WooCommerce site, navigate to Example:
https://wpml.loc/wp-admin/post.php?post=1&action=edit&return_url=https://wpml.loc/

### WordPress Environment

No need for more details, the bug is clear and located in the code.

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### Extra notes
- This bug was found in the context of a full automatic E2E test campaign between WC 9.0.0-rc.1 and WPML and WooCommerce Multilingual.
- This bug is causing critical issues and we expect it to be fixed before the release of WC 9.0.0.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #48154 .

We currently redirect users to `redirect_url` without checking for `do_update_woocommerce`. The redirection should only occur when both `do_update_woocommerce` and `redirect_url` params are present. 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site WITHOUT WooCommerce.
2. Install and activate an [older version](https://downloads.wordpress.org/plugin/woocommerce.8.6.1.zip) of WooCommerce. 
3. Build this branch locally by running `pnpm run build:zip` in `plugins/woocommerce`
4. Upload the zip file and activate it.
5. Test the bug by visiting https://JN-SITE-URL/wp-admin/post.php?post=1&action=edit&return_url=https://JN-SITE-URL/wp-admin/admin.php?page=wc-admin
6. Redirection should not happen.
7. Go to `WooCommerce -> Home`
8. Confirm the D.B update notice.
9. Click on the update button.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Fix db update notice redirection bug where it redirects without checking for db update action.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by adrianduffell — APPROVED** [review]
Nice fix and tested well, thanks @moon0326! 🚀

File: plugins/woocommerce/changelog/48163-fix-db-update-redirect-issue [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Fix db update notice redirection bug where it redirects without checking for db update action.  </details>  <details>  <summary>Changelog Entry Comment</summary>
```
Comment (chihsuan) [inline plugins/woocommerce/changelog/48163-fix-db-update-redirect-issue:4]:
```suggestion
Fix db update notice redirection bug where it redirects without checking for db update action.
```

**Review by chihsuan — APPROVED** [review]
Thanks for the fix! Looks good. 👍 

Just left a comment on changelog.

**Comment by adrianduffell** [comment]
Merging on behalf of @moon0326

**Pull request merged**